libs/libarchfpga/src/read_xml_arch_file.cpp

```diff
@@ -2709,31 +2728,61 @@ static void ProcessComplexBlocks(pugi::xml_node Node,
 
 	CurType = Node.first_child();
 	while (CurType) {
-		check_node(CurType, "pb_type", loc_data);
+		check_node(CurType, "tile", loc_data);
 
 		/* Alias to current type */
 		Type = &(*Types)[i];
 
 		/* Parses the properties fields of the type */
-		ProcessComplexBlockProps(CurType, Type, loc_data);
+		ProcessTileProps(CurType, Type, loc_data);
 
 		ret_pb_type_descriptors = pb_type_descriptors.insert(
 				pair<string, int>(Type->name, 0));
 		if (!ret_pb_type_descriptors.second) {
 			archfpga_throw(loc_data.filename_c_str(), loc_data.line(CurType),
-					"Duplicate pb_type descriptor name: '%s'.\n", Type->name);
+					"Duplicate tile descriptor name: '%s'.\n", Type->name);
 		}
 
-		/* Load pb_type info */
-		Type->pb_type = new t_pb_type;
-		Type->pb_type->name = vtr::strdup(Type->name);
-		ProcessPb_Type(CurType, Type->pb_type, nullptr, timing_enabled, arch, loc_data);
-		Type->num_pins = Type->capacity
-				* (Type->pb_type->num_input_pins
-						+ Type->pb_type->num_output_pins
-						+ Type->pb_type->num_clock_pins);
-		Type->num_receivers = Type->capacity * Type->pb_type->num_input_pins;
-		Type->num_drivers = Type->capacity * Type->pb_type->num_output_pins;
+		Type->index = i;
+		Type->available_tiles_indices.insert(i);
+
+		TypeMap->insert(std::make_pair(Type->name, Type));
```

Comment:
Should check return value, and error on duplicate type names?

Reply:
Done